rfcs/0168-feature-parameter-names.md

```diff
@@ -0,0 +1,559 @@
+has neither, and this RFC strives to rectify it.
+
+# Detailed design
+[design]: #detailed-design
+
+When upstream build system has options to enable or disable a optional features
+or configure some compile-time parameter, like path to default mailbox or
+initial size of hash map, derivation function (for short, package) MAY expose
+compile-time options via feature parameters that match
+`^(with|conf|enable)_[a-z_0-9]$` regular expression. For short, we will refer
+to them as `with`-parameter, `conf`-parameter and `enable`-parameter correspondingly,
+collectively as `feature parameters`.
+
+Note that this RFC does not establish any requires whether package should or
+should not have any feature parameters, this decision is left at descretion of
+the package maintainer(s). This RFC only concerns itself with the naming of the
+feature flags. Further text will use wording "corresponding feature parameter"
+with implicit caveat "provided that package maintainer decided to export
+underlying build system configuration option".
+
+Feature parameters MUST only be used according to the rules outlined in this
+RFC. They MUST NOT be used for any other purpose.
+
+Boolean feature parameter MUST be either `with`-parameter or `enable`-parameter.
+Non-boolean (e.g. string or integer) feature parameter MUST be an `conf`-parameter.
+
+Feature paramter that incurs additional runtime dependency MUST be and `with`-parameter.
+
+<details><summary>Example</summary>
+
+```
+{ lib, stdenv, mpfr, with_mpfr ? true }:
+
+stdenv.mkDerivation {
+  # ... snip ...
+  configureFlags = lib.optionals with_mpfr ["--with-mpfr"];
+  buildInputs = lib.optionals with_mpfr [ mpfr ];
+  # ... snip ...
+}
+```
+
+</details>
+
+As special provision, if optional vendored dependency is exposed by feature
+parameter, it MUST be `with`-parameter.
+
+If feature parameter does not incur extra runtime dependencies, corresponding
+feature parameter MUST be `enable`-parameter and its name SHOULD correspond to
+the upstream naming, unless is causes major inconsistency with other packages.
+Consistency of feature parameters across nixpkgs is more important than
+matching to the names used by particular upstream. What constitutes a major
+inconsistency is left at discretion of the package maintainer(s).
+
+<details><summary>Example</summary>
+
+If upstream uses `--enable-translations` to denote support for NLS,
+corresponding feature parameter SHOULD be `enable_nls`, since it is much more
+common term for enabling programs to produce output in non-English languages.
+
+</details>
+
+<details><summary>Rationale</summary>
+
+This distinction between `with` and `enable` feature parameters is based on Autoconf naming conventions.
+
+[Autoconf ENABLE](https://www.gnu.org/software/autoconf/manual/autoconf-2.66/html_node/Package-Options.html)
+[Autoconf WITH](https://www.gnu.org/software/autoconf/manual/autoconf-2.66/html_node/Package-Options.html)
```

Comment:
Is it intended to be the same link two times with two different labels?

Reply:
Nope, it is bug. Fixed.